.github/workflows/vercel-deploy.yml

```
@@ -0,0 +1,155 @@
+      
+    - name: 🚀 Deploy to Vercel (Preview)
+      uses: amondnet/vercel-action@v25
+      with:
+        vercel-token: ${{ secrets.VERCEL_TOKEN }}
+        vercel-org-id: ${{ secrets.VERCEL_ORG_ID }}
+        vercel-project-id: ${{ secrets.VERCEL_PROJECT_ID }}
+        scope: ${{ secrets.VERCEL_ORG_ID }}
+        
+    - name: 🧪 Test Preview Deployment
+      if: success()
+      run: |
```

actionlint (1.7.7)

property "deploy" is not defined in object type {}

(expression)